src/JsonWriter.cpp

```diff
@@ -0,0 +1,104 @@
+#include "JsonWriter.hpp"
+#include <stdio.h>
+
+namespace t2
+{
+
+void JsonWriteInit(JsonWriter* writer, MemAllocHeap* heap)
+{
+    writer->m_Heap = heap;
+    BufferInitWithCapacity(&writer->m_Buffer, heap, 200);
```

Comment:
don't we want most JsonWriters to use scratch memory?

Reply:
Maybe, but the Buffer APIs only take MemAllocHeap allocators, not MemAllocLinear...

Reply:
If we want to use scratch memory, what I could do is change the JsonWriter to store a linked-list of chunks. Is it worth it for this PR, though?

Reply:
Turns out it wasn't that involved and actually makes the scope of the PR smaller \m/

7582c8e7ebab9d279b14cfdb19f30075480cff9f